sklearn/linear_model/ridge.py

```diff
@@ -327,7 +327,7 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         y = check_array(y, dtype=np.float64, ensure_2d=False, order='F')
     else:
         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],
-                        dtype=np.float64)
+                        dtype='numeric')
```

Comment:
I can test it again. However, as I understand it, the goal of this PR is:

- to ensure dtype consistency between input (`X`) and output (`self.coef_`),
- to make sure the computations are using X with its orginal dtype (to reduce memory footprint as asked in #8769).

Wouldn't sticking with `np.float64` cancel the second bullet point?

Just for reference: `test_ridge.py` does not appear to make use of unit tests, so the way I test the code is by adding the following lines at the end of my local version of `test_ridge.py`:
```
if __name__ == "__main__":
    test_dtype_consistency()
```